reactor-core/src/main/java/reactor/core/publisher/FluxMetrics.java

```diff
@@ -183,7 +184,7 @@
 		this.tags = nameAndTags.getT2();
 
 		if (registry == null) {
-			this.registryCandidate = Metrics.globalRegistry;
+			this.registryCandidate = (MeterRegistry) Metrics.getUnsafeRegistry();
```

Comment:
Since we're in Java 8+ world, maybe we should return `Optional< MeterRegistry>` from `Metrics.getUnsafeRegistry` and use `this.registryCandidate = Metrics.getUnsafeRegistry().orElse(registry);`?

Reply:
yeah, but the whole stuff falls apart the minute the API expose a Micrometer type and users refer to the `Metrics` class 😢

Reply:
actually, as long as one doesn't try to "resolve" the `Optional` (eg. via `get()`/`orElse`/anything that would return a `MeterRegistry`), this suggestion seems to work, thanks to type erasure.

Reply:
right, that doesn't seem to be a problem with my current VM but it could prove to be one with a different VM (since nothing in the JVM spec seems to prevent eager-loading of classes of fields that have not yet been used). so I should probably be extra-careful and only have a `static Object registryCandidate` field.

Reply:
(note that the same way the `MeterRegistry` field seems fine on `1.8.0_162`, the `Optional` version seems also fine. but that sounds like a dangerous experiment to rely on).